blog/content/post/2020-03-27-trenchboot-nlnet-lz.md

```diff
@@ -0,0 +1,419 @@
+particular fork and branch. Every change appears there, so it is definitely
+place where you will find up-to-date firmware. I won't describe coreboot itself
+here. What you need to know, it is first code which is running during boot
+process. It initialize all hardware components on platform.
+
+#### GRUB
+
+When coreboot ends its work, a bootloader comes in. We use **GRUB**. Its task is
+to boot operating system. Normally, if NixOS is installed on hard disk, GRUB
+loads it to RAM. Being more precise, *Linux kernel* and *initrd* are being
+loaded. However, that operation is slightly different when we enable DRTM. As it
+is main goal of our project, let's take a look a little bit closer on it.
+
+We prepared extension module called `slaunch`. When DRTM is enabled, `slaunch`
+executes AMD's dedicated machine instruction `SKINIT`. Treat is 'pre-DRTM'
+operation. It prepares all components and jump to secure loader - `Landing
+Zone`.
+
+#### Landing Zone (LZ)
+
+LZ has 2 main tasks: measure Linux kernel and then run it. If you wonder, when
+measurements are done, then you have just find out. Without LZ, DRTM doesn't
+exist. Although it seems inconspicuous, it does fundamental job in secure boot
+process.
+
+#### Linux kernel
+
+Linux kernel is a core of all Linux based operating systems. Actually, it is
+operating system without any additional applications. It can be modified to
+given platform by including and excluding particular drivers, called modules. In
+general, it is done to limit size of kernel image and adjust its performance.
+
+#### initrd
+
+initrd stands for *initial ramdisk*. It is image used by bootloader to mount
+initial file system in memory. Also, that image contains modules which will be
+loaded by Linux kernel.
+
+## Bring it all together
+
+I have mentioned and described all parts of system which you need to know, to
+understand TrenchBoot and DRTM. But how they are related and how to build such
+system? I said that package manager in NixOS is very important and we will
+commonly use it. And now I want to show you how.
+
+First, we have 'clean' NixOS operating system installed on hard drive. It has
+default Linux kernel and initrd. From GRUB we can boot to it and... that's it.
+No DRTM, no measurements and no secure. But, with custom package management, we
+can **enable DRTM** and **prepare all components** directly from operating
+system. That's what we did. We have prepared, in separate packages, all elements
+mentioned above:
+
+- GRUB-trenchboot - it updates GRUB, so it boot `slaunch` and in consequence LZ
+- LZ-trenchboot - it adds LZ which measures and runs Linux kernel (enables DRTM)
+- linux-kernel-trenchboot - customized NixOS kernel
+- initrd-trenchboot - customized initrd
+
+Installing those packages will create a target system with enabled DRTM. How to
+do this in practice and verify if it really works will be discussed in next
+article! So definitely, wait for next post to use this knowledge in practice.
```

Comment:
It is a little bit weird to encourage somebody to do something secure, but wait a few days to know what it will do to your machine... What should we do about it? @pietrushnic cc

Reply:
As both articles will be published in the same day, I rephrase it to indicate to next article immediately.